ptop/main.py

```diff
@@ -28,23 +28,127 @@ def main():
+                            action='store',
+                            type=float,
+                            required=False,
+                            help=
+                            '''
+                                Input sensor
+        						update interval in
+        						milli seconds less than 1000.
+        						For example 500
+                            ''')
+
+        parser.add_argument('-proc',
+                            dest='proc',
+                            action='store',
+                            type=float,
+                            required=False,
+                            help=
+                            '''
+                                Input sensor
+        						update interval in
+        						milli seconds less than 1000.
+        						For example 500
+                            ''')
+
+        parser.add_argument('-sys',
+                            dest='sys',
+                            action='store',
+                            type=float,
+                            required=False,
+                            help=
+                            '''
+                                Input sensor
+        						update interval in
+        						milli seconds less than 1000.
+        						For example 500
+                            ''')
+        
         parser.add_argument('-v',
                             action='version',
                             version='ptop {}'.format(__version__))
 
         results = parser.parse_args()
+
         if results.theme:
             theme = results.theme
         else:
             theme = 'elegant'
+
+        if results.cpu:
+            cput = float(results.cpu)
+            if cput>1000:
+            	cput = 1000
+        else:
+            cput = 500
+
+        if results.disk:
+            diskt = float(results.disk)
+            if diskt>1000:
+            	diskt = 1000
+        else:
+            diskt = 500
+
+        if results.mem:
+            memt = float(results.mem)
+            if memt>1000:
+            	memt = 1000
+        else:
+            memt = 500
+
+        if results.proc:
+            proct = float(results.proc)
+            if proct>1000:
+            	proct = 1000
+        else:
+            proct = 500
+
+        if results.sys:
+            syst = float(results.sys)
+            if syst>1000:
+            	syst = 1000
+        else:
+            syst = 500
+
+        sensor_rate = [cput, diskt, memt, proct, syst]
         # app wide global stop flag
         global_stop_event = threading.Event()
 
-        s = Statistics(SENSORS_LIST,global_stop_event)
+        s = Statistics(SENSORS_LIST,global_stop_event,sensor_rate)
         # internally uses a thread Job 
         s.generate()
         logger.info('Statistics generating started')
-        app = PtopGUI(s.statistics,global_stop_event,theme)
+        app = PtopGUI(s.statistics,global_stop_event,theme,sensor_rate)
```

Comment:
Why do we require it in here, only passing to `Statistics` class will do the job.

Reply:
The sensor gets the data in specified interval, but it's not displayed in GUI accordingly. To do the same the update rate is sent to PtopGUI.

Reply:
We can assume that the GUI refresh rate is minimum compared to other sensor response times but its fine as I see you are taking minimum of all. Should we make a gitter channel of all interested folks interested in contributing ?

Reply:
Sure.